lib/schema.js

```diff
@@ -428,21 +464,41 @@ class Metaschema {
           value = def(value);
           obj[field] = value;
         } else {
-          value = this.createInstance(def, value);
+          if (def.category) value = this.createInstance(def.category, value);
+          if (def.domain) value = this.createDomainInstance(def.domain, value);
           if (value !== null) obj[field] = value;
           else return null;
         }
       }
-      for (const prop of required) {
-        if (obj[prop] === null || obj[prop] === undefined) {
-          return null;
+      for (const prop of properties) {
+        const def = definition[prop];
+        if (!obj.hasOwnProperty(prop)) {
+          if (def.default) {
+            let value = null;
+            value = this.createDomainInstance(def.domain, def.default);
+            if (value === null) return null;
+            obj[prop] = value;
+          } else if (def.required) {
+            return null;
+          }
         }
       }
       return obj;
     };
     return factory;
   }
 
+  // Add a Class property to a definition of Enum and Flags
+  [classifyDomains]() {
+    for (const definition of this.domains.values()) {
+      const domainKind = extractDecorator(definition);
+      const domainConstructor = domainConstructors[domainKind];
+      if (!domainConstructor) continue;
```

Comment:
IMO, it'd be better to just use `if` without continue, the branch will only have 2 lines.